python/dgl/transform.py

```diff
@@ -404,6 +405,56 @@ def laplacian_lambda_max(g):
                                       return_eigenvectors=False)[0].real)
     return rst
 
+def coalesce_metapath(g, metapath):
+    """Transform a heterogeneous graph into either a homogeneous graph or a unidirectional
+    bipartite graph so that metapaths are coalesced into edges.
```

Comment:
How about `metapath_reacheable_graph`? Also, for the docstring, I won't mention homogeneous or bipartite in the first sentence. How about "Return a graph where the neighbors of a node are the nodes reachable by the given metapath in the input graph"?